### PR TITLE
Enemy now has attributes, can be affected by spells

### PR DESCRIPTION
**Pull request by mutantmell**

### CODE REVIEW

File: Assets/Src/State/Model/Attributes.cs [hunk 1]
```diff
@@ -24,8 +24,15 @@ public static Attributes FromBase(BaseAttributes baseAttributes)
         }
 
         public BaseAttributes BaseAttributes { get; private set; }
-        public int Damage { get; private set; }
```
Comment (akalman) [inline Assets/Src/State/Model/Attributes.cs]: should still be private

File: Assets/Src/State/Model/Attributes.cs [hunk 2]
```diff
@@ -24,8 +24,15 @@ public static Attributes FromBase(BaseAttributes baseAttributes)
...
+        public int Damage { get; set; }
         public Modifier[] Modifiers { get; private set; }
+        public int Heath
+        {
+            get
+            {
+                return BaseAttributes.BaseHealth - Damage;
```
Comment (akalman) [inline Assets/Src/State/Model/Attributes.cs]: do we want to keep track of current health or damage?

I can't think of a meaningful difference, but maybe it affects how we model temporary health or something like that?

File: Assets/src/Game/StateTransforms/SpellCastTransform.cs [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+                playerOne: GetUpdatedState(sce.casters, Players.PlayerOne),
+                playerTwo: GetUpdatedState(sce.casters, Players.PlayerTwo)
+            );
+
+            var effected = levelState.Enemies.Where(e => Vector3.Distance(e.Position, sce.center) < sce.spell.Radius);
+            foreach (var e in effected)
+            {
+                Debug.Log("spos: " + sce.center);
+                Debug.Log("epos: " + e.Position);
+                Debug.Log("dist: " + Vector3.Distance(e.Position, sce.center));
```
Comment (akalman) [inline Assets/src/Game/StateTransforms/SpellCastTransform.cs]: we should think about what our logging strategy is

my current thought is to have a log statement per transform

Comment (mutantmell) [reply Assets/src/Game/StateTransforms/SpellCastTransform.cs]: That sounds like a good plan moving forward.

File: Assets/Src/State/Model/Attributes.cs (hunk 2 above)
Comment (mutantmell) [reply Assets/Src/State/Model/Attributes.cs]: The reason I chose to keep track of current damage is to keep health immutable, and with that goal it made more sense to just count up from 0.

File: Assets/Src/State/Model/Attributes.cs (hunk 1 above)
Comment (mutantmell) [reply Assets/Src/State/Model/Attributes.cs]: same

Comment (mutantmell) [reply Assets/Src/State/Model/Attributes.cs]: Damage was marked public to allow the SpellCastTransform to mark damage.  There a different way we want to do this for now?

File: Assets/Src/CopyConstructors.cs [hunk 4]
```diff
@@ -65,4 +65,31 @@ public static class CopyConstructors
...
+            mode ?? enemyState.Mode,
+            position ?? enemyState.Position,
+            direction ?? enemyState.Direction,
+            attributes ?? enemyState.Attributes
+            );
+    }
+
+    public static Attributes Set(
+        this Attributes attributes,
+        int damage = -1,
```
Comment (akalman) [inline Assets/Src/CopyConstructors.cs]: not an `int?`?

File: Assets/Src/State/Model/Attributes.cs (hunk 2 above)
Comment (akalman) [reply Assets/Src/State/Model/Attributes.cs]: ok

**Pull request merged**